tests/test_pipeline_metadata.py

```
@@ -0,0 +1,182 @@
+    path = tmp_path / "chunks.jsonl"
+    path.write_text("\n".join(json.dumps(record) for record in records), encoding="utf-8")
+    return path
+
+
+def _run_pipeline(chunks_file: Path, *, collection: str = "default") -> tuple[int, dict]:
+    captured: dict = {}
+
+    class FakeEmbedder:
+        dim = 4
+
+        def embed(self, docs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `embed`

(ANN202)